config/routes.rb

```diff
@@ -59,23 +59,25 @@
   delete "/cart/:item_id", to: "cart#remove_item"
   post "/cart/update_quantity/:item_id", to: "cart#update_quantity"
 
-  #orders
-  get "/orders/new", to: "orders#new"
-  post "/orders", to: "orders#create"
-  get "/orders/:id", to: "orders#show"
-  get "/profile/orders", to: "orders#index"
-
   #users
   get "/login", to: "users#login"
   get '/logout', to: "users#logout"
+
+  get "/profile", to: "users#show"
   get "/register", to: "users#new"
   post "/users", to: "users#create"
-  get "/profile", to: "users#show"
-
   get "/profile/edit", to: "users#edit"
   post "/profile", to: "users#update"
+
   get "/password/edit", to: "users#edit_pass"
   post "/password", to: "users#update_pass"
 
+  #orders
+  get "/orders/new", to: "orders#new"
+  post "/orders", to: "orders#create"
+  get "/profile/orders/:id", to: "orders#show"
+  get "/profile/orders", to: "orders#index"
+  patch "/profile/orders/:id", to: "orders#update"
```

Comment:
This is for cancelling an order. Instead of destroy, using an update and setting the status to cancelled.

Reply:
![](https://media1.giphy.com/media/lAb6xs6tLoZ44/giphy.gif)